### PR TITLE
feat(foundation): add schema attribute definition parser

### PR DESCRIPTION
**Pull request by JakobVogelsang**
This is the second of three PRs that is aiming to simplify wizard-dialog definition for co-developers of OpenSCD as well as plug-in contributors. The basic idea behind the function `schemaAttributeDefinition` is to get the attributes' definition for a specific element. Say you want to manipulate the `name` attribute of the element `GSEControl` in a standard way this function will tell you if there are restrictions for this attribute like pattern, minLength, maxLength its base type and or enumerations.

### CODE REVIEW

File: test/unit/foundation/schema/attribute.test.ts [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+      expect(schemaAttributeDefinition('nonsence', schema2007B4)).to.be.empty);
+
+    describe('for element SampledValueControl', () => {
+      describe('and version 2003', () => {
+        const attributes = schemaAttributeDefinition(
+          'SampledValueControl',
+          schema2003
+        );
+
+        it('return the correct number of attributes ', () =>
```
Comment (danyill) [inline test/unit/foundation/schema/attribute.test.ts:33]: There is a few of these with a space at the end you might like to remove.

```suggestion
        it('return the correct number of attributes', () =>
```

File: test/unit/foundation/schema/attribute.test.ts [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+              .pattern
+          ).to.be.undefined);
+
+        it('return an optional minLength', () =>
+          expect(
+            attributes.find(attribute => attribute.name === 'name')?.restriction
+              .minLength
+          ).to.equal('1'));
+
+        it('return an otional maxLength', () =>
```
Comment (danyill) [inline test/unit/foundation/schema/attribute.test.ts:54]:
```suggestion
        it('return an optional maxLength', () =>
```

And a few with this typo

**Review by danyill — APPROVED** [review]
I did not do more than read the code but I can see that this would be helpful and is required. Thankyou :+1:

**Comment by JakobVogelsang** [comment]
Will be tackled within `open-scd-core` reporitory directly :)

**Pull request closed without merging**